src/main/java/com/xiaomi/infra/pegasus/rpc/async/MetaSession.java

```diff
@@ -148,16 +191,16 @@ void onFinishQueryMeta(final MetaRequestRound round) {
     boolean needSwitchLeader = false;
     rpc_address forwardAddress = null;
 
-    --round.maxQueryCount;
+    if (!round.roundWithTimeout) {
```

Comment:
for all meta request, the response contain `same` and `diff`, you should refactor it more general.

As for current processing, it is actually general. alll meta response  contain ERR_UNKNOWN ERR_SERVICE_NOT_ACTIVE ERR_FORWARD_TO_OTHERS, so it should name `onFinishMetaResquet`. for more meta requet, you should `if also` or `extends` to distinct. use `is also`, may be like this:
```java
void onFinishMetaRequest {
if (createApp) {
err = ERR_EXIST:
err= ERR_DROPING:
}

if (dropApp) {
......
} 

// follow is general 
f (op.rpc_error.errno == error_types.ERR_OK) {
      metaError = getMetaServiceError(op);
      if (metaError == error_types.ERR_SERVICE_NOT_ACTIVE) {
        needDelay = true;
      } else if (metaError == error_types.ERR_FORWARD_TO_OTHERS) {
}
.....
}

Reply:
Of course,  for simplicity, you can contain `general` handler.

Reply:
yes i should think more thoroughly....., and after i think a lot,   I find that the different logic which should be implemented for specific operator ought to be in the following inner code block
```
if (op.rpc_error.errno == error_types.ERR_OK) {
       metaError = getMetaServiceError(op);
 
       // do real operator's specific logic
      if (op instanceof create_app_operator) {
        if (metaError == error_types.ERR_BUSY_CREATING) {
          ((create_app_operator) (op)).get_response().getErr().errno = error_types.ERR_OK;
          round.callbackFunc.run();
          return;
        }
      } else if (op instanceof query_cfg_operator) {
        // do nothing now
      }

      // do operator's common logic
      if (metaError == error_types.ERR_SERVICE_NOT_ACTIVE) {
        needDelay = true;
      } else if (metaError == error_types.ERR_FORWARD_TO_OTHERS) {
        needSwitchLeader = true;
        forwardAddress = getMetaServiceForwardAddress(op);
      } else {
        round.callbackFunc.run();
        return;
      }
}
```
because the `outer rpc level`'s error is common for every operator, the difference is in the `error code in the return struct`

Reply:
Yeah, this is my opinion.  this pr you can just handle  

> // do operator's common logic 

and named the function as `onFinishMetaRequest`, leave `todo` for later